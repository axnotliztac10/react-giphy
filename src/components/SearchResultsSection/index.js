import React from 'react';

import GiphyCard from 'components/GiphyCard';
import { Text, Box, Flex, Link } from 'rebass';

export default class SearchResultsSection extends React.PureComponent {
  renderResult = (result, index) => {
    const { onAddRemoveFavorite } = this.props;

    return (
      <GiphyCard
        giphy={result}
        key= {index}
        onAddRemoveFavorite={onAddRemoveFavorite}
      />
    );
  }

  render() {
    const { results, onLoadMore } = this.props;

    if (!results.length) {
      return null;
    }

    return (
      <Box my={4}>
        <Text textAlign="center" fontSize={22}>Results:</Text>
        <Flex
          flexDirection="row"
          justifyContent="center"
          alignItems="flex-start"
          flexWrap="wrap"
        >
          {results.map(this.renderResult)}
          <Box width={1}>
            <Link fontSize={30} onClick={() => onLoadMore()}>Load more</Link>
          </Box>
        </Flex>
      </Box>
    );
  }
}
