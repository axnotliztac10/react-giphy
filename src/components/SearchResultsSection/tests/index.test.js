import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import SearchResultsSection from '../index';

describe('<SearchResultsSection />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <SearchResultsSection
        results={[{}]}
        onLoadMore={() => {}}
        onAddRemoveFavorite={() => {}}
      />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
