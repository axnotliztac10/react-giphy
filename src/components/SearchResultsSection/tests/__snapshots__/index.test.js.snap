// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultsSection /> should render correctly 1`] = `
<styled.div
  my={4}
>
  <Styled(styled.div)
    fontSize={22}
    textAlign="center"
  >
    Results:
  </Styled(styled.div)>
  <Styled(styled.div)
    alignItems="flex-start"
    flexDirection="row"
    flexWrap="wrap"
    justifyContent="center"
  >
    <GiphyCard
      giphy={Object {}}
      key="0"
      onAddRemoveFavorite={[Function]}
    />
    <styled.div
      width={1}
    >
      <Styled(styled.div)
        as="a"
        color="blue"
        fontSize={30}
        onClick={[Function]}
      >
        Load more
      </Styled(styled.div)>
    </styled.div>
  </Styled(styled.div)>
</styled.div>
`;
